include/redisserver.h

```diff
@@ -312,6 +379,28 @@ class RedisServer {
                                         std::vector<std::string> outputs)
                                          = 0;
 
+        /*!
+        *   \brief Run a script function in the database using the
+        *          specificed input and output tensors in a multi-GPU system
+        *   \param name The name associated with the script
+        *   \param function The name of the function in the script to run
+        *   \param inputs The names of input tensors to use in the script
+        *   \param outputs The names of output tensors that will be used
+        *                  to save script results
+        *   \param image_id index of the current image, such as a processor
+        *                   ID or MPI rank
+        *   \param first_gpu The first GPU to use with this script
+        *   \param num_gpus the number of gpus for which the script was stored
+        *   \throw RuntimeException for all client errors
+        */
+        virtual void run_script_multigpu(const std::string& name,
+                                         const std::string& function,
+                                         std::vector<std::string>& inputs,
+                                         std::vector<std::string>& outputs,
+                                         int image_id,
+                                         int first_gpu,
+                                         int num_gpus) = 0;
```

Comment:
I could see in the future ``first_gpu`` have a default value (i.e. ``0``).  Do you think it would make sense to put it after ``num_gpus`` so if we do that in the future we won't break the API?

Reply:
If we decide to do that we can also set num_gpus to default to 1 and not break the API